lisa/target.py

```diff
@@ -688,14 +692,6 @@ def _init_target(self, kind, name, workdir, device, host,
 
         target.connect(check_boot_completed=wait_boot, timeout=wait_boot_timeout)
 
-        # None as username means adb root will be attempted, but failure will
-        # not prevent from connecting to the target.
-        if kind == 'android' and username is None:
-            try:
-                target.adb_root(enable=True)
```

Comment:
looks like this should be turned into something like that, following the original devlib implementation (with the extra necessary import for AdbConnection):
```
conn = target.conn
if isinstance(conn, AdbConnection) and username is None:
	try:
		conn.adb_root(enable=enable)
	except Exception as e: # pylint: disable=broad-except
		logger.warning(f'"adb root" failed: {e}')
```

Reply:
Hopefully that will be enough. If devlib does not appreciate late rooting anymore, we could also try to connect to the target as root and bail out if it does not work when username is None (but I'd rather try this solution first, catching the "non rooted" error will be problematic and we will end up catching network error etc as well)

Reply:
Actually had a thought about that: intervening directly on `target.conn` is not really going to cut it. This attribute is thread local and new connections will be instantiated and destroyed as threads are used. Setting it for the main thread would provide a partial workaround but it's not really a fix unfortunately.